packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -2709,6 +2709,70 @@ class SizedOverflowBox extends SingleChildRenderObjectWidget {
 /// needed, prefer removing the widget from the tree entirely rather than
 /// keeping it alive in an [Offstage] subtree.
 ///
+/// {@tool dartpad --template=stateful_widget_material}
+///
+/// This example shows a [FlutterLogo] widget when the `_offstage` member field
+/// is false, and hides it without any room in the parent when it is true. The
+/// [Size] of the logo without bringing it on screen will be measured.
+///
+/// ```dart
+/// GlobalKey _key = GlobalKey();
+/// bool _offstage = true;
+/// Size _invisibleLogoSize;
+///
+/// void _changeOffstageValue() {
+///   setState(() {
+///     _offstage = !_offstage;
+///     _invisibleLogoSize = null;
+///   });
+/// }
+///
+/// void _getInvisibleLogoSize() {
+///   setState(() {
+///     final RenderBox renderLogo = _key.currentContext.findRenderObject();
+///     final sizeLogo = renderLogo.size;
+///     _invisibleLogoSize = sizeLogo;
```

Comment:
Can you only wrap this last line in the setState and move everything else out of it? Only this last line is actually changing state. (It doesn't matter much here, but if one of the methods called here would do async work, it would matter).